### DISCUSSION TITLE
Attach attributes to a `<li>` element?

### DISCUSSION THREAD

**Discussion by tmke8** [jgm/djot#183] [General]
If I write this
```
{#main}
- A list item.
```
the block attribute gets attached to the `<ul>` element:
```html
<ul id="main">
<li>
A list item.
</li>
</ul>
```
But what if I want to attach it to the `<li>` element? Is there a way?

EDIT: 

the use case is that I want to reference a deeply nested list item:
```
- item 1
- item 2

  - nested item

    - nested again
    {#my-ref}
    - important information

  - another nested item

- item 3

[See here.](#my-ref)
```

**Comment by jgm** [comment]
Currently not.  Any attribute before a list item gets treated as an attribute on the whole list (and a new list starts if you were already in one).

Open to sugestions about how this could be done in a consistent way.

**Reply by bpj** [reply]
A blank line between the attributes and the first list item if the attributes are supposed to apply to the list would probably be the simplest.

(Note to the anti-blank-lines crowd: I already know that we disagree!)

**Reply by jgm** [reply]
But in a loose list you'll have blank lines between all items.

**Reply by tmke8** [reply]
Perhaps like this?

```
- beginning of list
- {#foo .bar .baz key="my value"} A list item.
- end of list
```
It seems to me there is no risk of this being confused with an inline attribute because there is nothing in front of it.

**Reply by jgm** [reply]
Yes, maybe that's the solution.
Want to create an issue suggesting this?

**Reply by tmke8** [reply]
Done.

**Reply by bpj** [reply]
> But in a loose list you'll have blank lines between all items.

Perhaps this is clearer than words?

``````djot
{#list}

{#item-1}
-   Foo

{#item-2}
-   Bar
``````
